src/forwardproblem.cpp

```diff
@@ -615,41 +574,37 @@ void ForwardProblem::getDataSensisFSA(int it) {
      * @param[in] it index of current timepoint @type int
      */
 
-    if (!(std::isinf(rdata->ts[it]))) {
-        if (rdata->ts[it] > model->t0()) {
-            solver->AMIGetSens(&(t), &sx);
-        }
+    if (!std::isinf(rdata->ts[it]) && rdata->ts[it] > model->t0()) {
+        solver->AMIGetSens(&(t), &sx);
     }
     
-    if (model->nx > 0) {
-        for (int ix = 0; ix < model->nx; ix++) {
-            for (int ip = 0; ip < model->nplist(); ip++) {
-                rdata->sx[(it * model->nplist() + ip) * rdata->nx + ix] =
-                sx.at(ix,ip);
-            }
+    for (int ix = 0; ix < model->nx; ix++) {
+        for (int ip = 0; ip < model->nplist(); ip++) {
+            rdata->sx[(it * model->nplist() + ip) * rdata->nx + ix] =
+                    sx.at(ix,ip);
         }
     }
 
-    for (int iy = 0; iy < model->nytrue; iy++) {
-        if (edata) {
+    if(edata) {
+        for (int iy = 0; iy < model->nytrue; iy++) {
             if (isNaN(edata->sigmay[it * rdata->nytrue + iy])) {
-                model->fdsigma_ydp(it, rdata);
+                // @ REVIEW: it seems redundant to call this for every iy, should be only called once per it, not?
```

Comment:
yes, also looks redundant to me, actually, most of this code block seems redundant with the current model->fdsigma_ydp implementation. I wanted to move the respective code to the model class anyways as its more consistent with the other code to have the responsability there.

Reply:
Postponed to #332